### PR TITLE
Fix SEO issues to impove the Ahrefs' Health Score

### PR DESCRIPTION
**Pull request by nqst**
Made some changes that should improve our SEO [Health Score](https://help.ahrefs.com/en/articles/1424673-what-is-health-score-and-how-is-it-calculated-in-ahrefs-site-audit) provided by Ahrefs. As of Dec 2023, the score was 87%. 

**Changes:**

- Fixed broken links.
- Added `og:image` and other Open Graph & Twitter tags.
- Added meta descriptions for the website pages. I didn't add the descriptions to all blog posts, but added an example one to the most recent post.
- Fixed the [Noindex page in sitemap](https://help.ahrefs.com/en/articles/2754358-what-does-the-noindex-page-in-sitemap-issue-in-site-audit-mean) issue.
- Optimized a heavyweight image.

**Unrelated change:** running `yarn inject:protocol` added some changes to the protocol that weren't added before.

### CODE REVIEW

**Review by nickrttn — APPROVED** [review]
That's amazing @nqst 👏 

Is the example blog post description ok to go live with or should someone (@Acconut?) write descriptions for each post?

**Comment by nqst** [comment]
> Is the example blog post description ok to go live? 

Yeah, that's just an excerpt from the blog post introduction, so it should be good to go.

> Should someone write descriptions for each post?

Ideally, yes, although that's a secondary priority compared to adding descriptions to the homepage and the pages that we have in the navbar.

**Review by kvz — APPROVED** [review]
💯💯💯💯
✨✨💖✨
💯💯💯💯

File: scripts/inject-protocol.sh [hunk 1]
```diff
@@ -32,6 +32,7 @@ function fetch_protocol {
 ---
 # This file is generated by _scripts/inject-protocol.sh. Do not modify it manually!
 title: Resumable upload protocol ${slug}
+description: "The Tus protocol specification: Detailing the interactions between Tus clients and servers for reliable and resumable file uploads."
```
Comment (Acconut) [inline scripts/inject-protocol.sh:35]: We do use the capitalized `Tus` on transloadit.com, but on tus.io we prefer `tus` for now :)

File: src/content/protocols/0.1.x.md [hunk 2]
```diff
@@ -1,6 +1,9 @@
 ---
 # This file is generated by _scripts/inject-protocol.sh. Do not modify it manually!
 title: Resumable upload protocol 0.1.x
+description:
+  'The Tus protocol specification: Detailing the interactions between Tus
```
Comment (Acconut) [inline src/content/protocols/0.1.x.md:5]:
```suggestion
  'The tus protocol specification: Detailing the interactions between tus
```

File: src/content/protocols/0.2.x.md [hunk 3]
```diff
@@ -1,6 +1,9 @@
 ---
 # This file is generated by _scripts/inject-protocol.sh. Do not modify it manually!
 title: Resumable upload protocol 0.2.x
+description:
+  'The Tus protocol specification: Detailing the interactions between Tus
```
Comment (Acconut) [inline src/content/protocols/0.2.x.md:5]:
```suggestion
  'The tus protocol specification: Detailing the interactions between tus
```

File: src/content/protocols/1.0.x.md [hunk 4]
```diff
@@ -1,6 +1,9 @@
 ---
 # This file is generated by _scripts/inject-protocol.sh. Do not modify it manually!
 title: Resumable upload protocol 1.0.x
+description:
+  'The Tus protocol specification: Detailing the interactions between Tus
```
Comment (Acconut) [inline src/content/protocols/1.0.x.md:5]:
```suggestion
  'The tus protocol specification: Detailing the interactions between tus
```

File: scripts/inject-protocol.sh (hunk 1 above)
Comment (Acconut) [inline scripts/inject-protocol.sh:35]:
```suggestion
description: "The tus protocol specification: Detailing the interactions between tus clients and servers for reliable and resumable file uploads."
```

File: src/pages/blog/index.astro [hunk 5]
```diff
@@ -8,6 +8,7 @@ import { getFirstParagraphContent } from '@/lib/getFirstParagraph'
 import { getCollection, getEntries } from 'astro:content'
 
 const title = 'Blog'
+const description = 'Explore our Blog for the latest updates and insights on resumable file uploading technology and protocol developments.'
```
Comment (Acconut) [inline src/pages/blog/index.astro:11]:
```suggestion
const description = 'Explore our blog for the latest updates and insights on resumable file uploading technology and protocol developments.'
```

File: src/pages/demo.mdx [hunk 6]
```diff
@@ -1,6 +1,9 @@
 ---
 layout: ../layouts/markdown.astro
 title: Resumable File Upload Demo
+description:
+  'Experience the Tus protocol in action with our live demo, showcasing
```
Comment (Acconut) [inline src/pages/demo.mdx:5]:
```suggestion
  'Experience the tus protocol in action with our live demo, showcasing
```

File: src/pages/faq.md [hunk 7]
```diff
@@ -1,6 +1,9 @@
 ---
 layout: ../layouts/markdown.astro
 title: FAQ
+description:
+  'Find answers to your questions about Tus. Get insights on how Tus facilitates
```
Comment (Acconut) [inline src/pages/faq.md:5]:
```suggestion
  'Find answers to your questions about tus. Get insights on how tus facilitates
```

File: src/pages/implementations.astro [hunk 8]
```diff
@@ -4,6 +4,8 @@ import ListItem from '@/components/ImplementationListItem.astro'
 
 import { getCollection } from 'astro:content'
 
+const description = 'Explore real-life Tus protocol implementations in various languages and platforms, showcasing its versatility and efficiency in file uploads.'
```
Comment (Acconut) [inline src/pages/implementations.astro:7]:
```suggestion
const description = 'Explore real-life tus protocol implementations in various languages and platforms, showcasing its versatility and efficiency in file uploads.'
```

File: src/pages/index.astro [hunk 9]
```diff
@@ -13,6 +13,7 @@ import BlogPosts from '@/components/Home/BlogPosts.astro'
 import PageFooter from '@/components/PageFooter.astro'
 
 const title = 'tus - resumable file uploads'
+const description = 'Tus is the open protocol standard for resumable and reliable file uploads across the web, facilitating efficient and seamless file transfer experiences.'
```
Comment (Acconut) [inline src/pages/index.astro:16]:
```suggestion
const description = 'tus is the open protocol standard for resumable and reliable file uploads across the web, facilitating efficient and seamless file transfer experiences.'
```

File: src/pages/support.mdx [hunk 10]
```diff
@@ -1,6 +1,10 @@
 ---
 layout: ../layouts/markdown.astro
 title: Support
+description:
+  'Explore community and commercial support options for Tus. Engage with our
+  community for protocol guidance, or opt for advanced support for your Tus
```
Comment (Acconut) [inline src/pages/support.mdx:6]:
```suggestion
  'Explore community and commercial support options for tus. Engage with our
  community for protocol guidance, or opt for advanced support for your tus
```

**Review by Acconut — CHANGES_REQUESTED** [review]
That's a great change, thank you very much, Alex!

**Review by Acconut — APPROVED** [review]

**Pull request merged**